frontend/src/components/Header.vue

```diff
@@ -1,69 +1,152 @@
-    >
-      <b-nav-item
-        v-for="link in links"
-        :to="link.to"
-        :key="link.text"
+  <div class="header">
+    <a
+      class="logo-container"
+      href="https://www.pah.org.pl/"
+      target="_blank">
+      <img
+        class="logo"
+        :src="logo" >
+    </a>
+    <nav class="menu">
+      <b-nav
+        fill
       >
-        {{ link.text }}
-      </b-nav-item>
-      <LoginStatus v-bind="user"/>
-    </b-nav>
+        <b-nav-item
+          v-for="link in links"
+          :to="link.to"
+          :key="link.text"
+        >
+          {{ link.text }}
+        </b-nav-item>
 
+        <b-nav-item
+          v-if="user"
+          to="/login"
+          key="logout"
+          class="username"
+        >
+          {{ $t('login.user', { username: user.username }) }}
+        </b-nav-item>
+
+        <b-nav-item
+          v-else
+          to="/login"
+          key="login"
+        >
+          {{ $t('common.login') }}
+        </b-nav-item>
+      </b-nav>
+    </nav>
   </div>
 </template>
 
 <script>
 
 import { mapState } from 'vuex';
-import LoginStatus from './LoginStatus.vue';
-import Language from './Language.vue';
+import { PL, languages } from '../main';
 
 export default {
   name: 'Header',
-  components: {
-    LoginStatus,
-    Language,
-  },
   computed: {
     ...mapState(['user']),
+    isLoginPage() {
+      return this.$route.name === 'Login';
+    },
+    logo() {
+      /* eslint-disable */
+      return this._i18n.locale === languages[PL]
+        ? require('../assets/logo_pl.svg')
+        : require('../assets/logo_en.svg');
+      /* eslint-enable */
+    },
   },
   data() {
     return {
-      links: [
-        {
-          text: this.$t('common.home'),
-          to: '/',
-        },
-        {
-          text: this.$t('common.new_route'),
-          to: '/route/',
-        },
-        {
-          text: this.$t('common.routes'),
-          to: '/routes/',
-        },
-      ],
+      links: this.navigation,
     };
   },
 };
 </script>
 
 <style scoped lang="scss">
-@import '../scss/variables';
-@import '../scss/mixins';
+@import '../scss/base';
 
 .header {
-  background: $blue;
+  @include p(3);
+  @include flex(row, center);
+
   text-align: center;
+  background: $white;
+  height: $header-height;
+
+  @include media-breakpoint-down (sm) {
+    height: $header-height-mobile;
+  }
+}
+
+.menu {
+  flex: 2;
+  padding-bottom: 20px;
+
+  @include media-breakpoint-down (sm) {
+    display: none;
+  }
+
+  & .nav {
+    & .nav-item {
+      max-width: 200px;
+    }
+
+    & .nav-item.username {
+      max-width: 400px;
+      flex-grow: initial;
+    }
+
+    & .nav-link:hover {
+      padding-bottom: 12px;
+      transition: 0.2s;
+    }
+
+    & > li > a {
+      min-height: 100px;
+      font-size: 20px;
+      min-width: 155px;
+      color: $white;
+      display: flex;
+      align-items: flex-end;
+    }
+
+    & > li:nth-child(1) {
+      background: $pah-color-1;
```

Comment:
We could have CSS classes with these colors and assign them in HTML.  I'm a fan of this solution but yours is equally good.